test/parallel/test-fs-read-position-validation.mjs

```diff
@@ -47,21 +43,23 @@ function testInvalidCb(code, position, callback) {
         () => fs.read(fd, { buffer, offset, length, position }, common.mustNotCall()),
         { code }
       );
+      assert.throws(
+        () => fs.read(fd, buffer, { offset, length, position }, common.mustNotCall()),
+        { code }
+      );
     } finally {
       fs.close(fd, common.mustSucceed(callback));
     }
   }));
 }
 
 // Promisify to reduce flakiness
-const testValidArrPositional = util.promisify(testValidCbPositional);
-const testValidArrNamedParams = util.promisify(testValidCbNamedParams);
+const testValidArr = util.promisify(testValidCb);
```

Comment:
Do we need the array part? I think we can add a `if (typeof allowedErrors === 'function') { callback = allowedErrors; allowedErrors = [] }`, that'd be more straightforward IMHO.
```suggestion
const testValid = util.promisify(testValidCb);
```

Reply:
Hmm, not a big fan of conditional "shifting" arguments. On a global scale, I'd prefer to have array-oriented version of `util.promisify()` (mainly because it doesn't comprehend functions with non-unary callbacks, unless there is a `customPromisifyArgs` workaround).
However, utility function is completely redundant here at this point. Returning Promise directly would be even more straightforward.